src/Components/WebAssembly/Server/src/Microsoft.AspNetCore.Components.WebAssembly.Server.csproj

```diff
@@ -7,17 +7,33 @@
     <HasReferenceAssembly>false</HasReferenceAssembly>
     <!-- This is so that we add the FrameworkReference to Microsoft.AspNetCore.App -->
     <UseLatestAspNetCoreReference>true</UseLatestAspNetCoreReference>
+
+    <!-- We're deliberately bundling assemblies as content files. Suppress the warning. -->
+    <NoWarn>$(NoWarn);NU5100</NoWarn>
   </PropertyGroup>
 
   <ItemGroup>
     <Compile Include="$(ComponentsSharedSourceRoot)\src\CacheHeaderSettings.cs" Link="Shared\CacheHeaderSettings.cs" />
-  </ItemGroup>
+    <Compile Include="$(SharedSourceRoot)\CommandLineUtils\Utilities\DotNetMuxer.cs" Link="Shared\DotNetMuxer.cs" />
 
-  <ItemGroup>
-    <Reference Include="Newtonsoft.Json" />
-
-    <!-- Used by ws-proxy sources only. Remove this once we're able to consume ws-proxy as a NuGet package. -->
-    <Reference Include="Mono.Cecil" />
+    <!-- Ensure debug proxy is built first, but don't create an actual reference, since we don't want its transitive dependencies. -->
+    <ProjectReference
+      Include="..\..\DebugProxy\src\Microsoft.AspNetCore.Components.WebAssembly.DebugProxy.csproj"
+      ReferenceOutputAssembly="false" />
   </ItemGroup>
 
+  <!-- Emit debug proxy binaries to output directory. This lets us launch it as a separate process while keeping the build output self-contained.  -->
+  <Target Name="IncludeDebugProxyBinariesAsContent" BeforeTargets="GetCopyToOutputDirectoryItems">
+    <ItemGroup>
+      <DebugProxyBinaries Include="..\..\DebugProxy\src\bin\$(Configuration)\$(DefaultNetCoreTargetFramework)\**" />
+      <ContentWithTargetPath Include="@(DebugProxyBinaries)"
```

Comment:
This feels very strange. Is there a reason the tool couldn't carry a copy of the proxy so it's not in the build output of the project?

Reply:
See option 2 in the PR description.